installers/msi-language/StateDeploy/CustomAction.cs

```diff
@@ -380,8 +380,33 @@ public InstallSequenceElement(string subCommand, string description)
         private static ActionResult run(Session session)
         {
             var sessionID = session.CustomActionData["SESSION_ID"];
+            var uiLevel = session.CustomActionData["UI_LEVEL"];
             var productVersion = session.CustomActionData["PRODUCT_VERSION"];
 
+            if (sessionID == "unset")
+            {
+                if (uiLevel != "2" /* no ui */ && uiLevel != "3" /* basic ui */)
```

Comment:
Yes, I guess that makes some of these changes obsolete.  Well, now it will be removed anyways.

But to answer your question.  UILEVEL==4, the `SetSessionID` immediate custom action was still executed.